test/treasure.track.spec.js

```diff
@@ -67,17 +68,34 @@ describe('Treasure Tracker', function () {
     })
 
     it('should set uuid to clientId if set manually', function () {
-      configuration.clientId = '123'
```

Comment:
I think we should have both tests here (passing clientId as configuration and, creating a generic uuid)